File: Java/src/com/dbc/algorithm/Tree/InvertBinaryTree.java
package com.dbc.algorithm.Tree;

import com.dbc.algorithm.ReferOffer.ErChaShuDeJingXiangLcof;

public class InvertBinaryTree {
    public class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;

        TreeNode(int x) {
            val = x;
        }
    }
    public TreeNode invertTree(TreeNode root) {
        if (root == null || (root.left == null && root.right == null)) {
            return root;
        }
        TreeNode temp = root.left;
        root.left = invertTree(root.right);
        root.right = invertTree(temp);

        return root;
    }
}

File: Python/tree/invert_binary_tree.py
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

def invertTree(self, root: TreeNode) -> TreeNode:
    def change(node: TreeNode) -> TreeNode:
        if not node or (not node.left and not node.right):
            return node
        
        temp = node.left
        node.left = change(node.right)
        node.right = change(temp)

        return node
    
    return change(root)